### PR TITLE
Add `.make` for simpler usage

### PR DESCRIPTION
**Pull request by shalvah-gs**
This PR makes it easier to get moving with NxtHttpClient. You often need a simple one-off client, and constructing a base class and all the attendant baggage isn't worth it. Now this is possible:

```rb
client = NxtHttpClient::Client.make do 
  configure do |config|
    config.base_url = 'www.httpstat.us'
    config.request_options.deep_merge!(
      followlocation: true
    )
  end
end

client.get('200')
```

I think going with required inheritance wasn't the best idea, as it complicates the usage (encourages inheritance hell, higher entry barrier), as well as the design and maintenance (we always have to consider unexpected mutations/shared state, internals are part of the API), but it's too late to overhaul that. This is a nice in-between.

[BE-46][]

[BE-46]: https://hellogetsafe.atlassian.net/browse/BE-46?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Aerdayne** [comment]
It's a great convenience feature, but I'm not sure that I see value in this in the context of our services. If there's need for a client to an external system, I think it's better to define it as a separate class. Can you please provide some current examples where we need one-off clients?

**Comment by shalvah-gs** [comment]
Sure.
1. Needed a simple client for the Aircall API [here](https://github.com/nxt-insurance/sales-service/blob/322ab6cfa93dd8638ebb4e625b557de0d4f14aa2/lib/tasks/webhooks.rake#L29).
2. Other places where we use an ad hoc client ([example](https://github.com/nxt-insurance/insurance-service/blob/ab9da4268df65fc78c5f4454951b0cebc63979da/lib/tasks/services/document_recipe_ingredients_copier.rb#L68))

But overall, why _require_ inheritance? HTTP clients are not shining examples of OOP. You just end up with unnecessary proliferation of classes.

**Comment by shalvah-gs** [comment]
Oh, I forgot to mention: nothing in this approach precludes one from having custom client classes as before. You can simply do `@client = NxtHttpClient::Client.make ...`. No harm done, and no spurious inheritance required.

**Comment by Aerdayne** [comment]
Thanks for the examples. I'm personally not satisfied with this gem's design, so I'm going to approve this since refactoring the whole thing and nxt_clients will be tedious. I just hope this doesn't get abused and we stick to defining shared clients in actual domain logic - for one-off rake tasks this is fine. Could you please mention this in the README or somewhere else?

**Comment by shalvah-gs** [comment]
@Aerdayne I already have this in the README:

> This is good when you need a one-off client to make some quick API calls. 
> However, you can go further by creating custom client classes with shared configuration,
> and customizing them as needed.

How's that?

File: README.md [hunk 1]
```diff
@@ -17,98 +16,97 @@ gem 'nxt_http_client'
...
+
+client.get('200')
 ```
 
-In order to setup a shared configuration you would therefore setup a client base class. The configuration and any
-response handler or callbacks you setup in your base class are then inherited to your concrete client implementations.
+This is good when you need a one-off client to make some quick API calls. 
+However, you can go further by creating custom client classes with shared configuration,
+and customizing them as needed.
+For example, you could have a base client for interacting with a specific service:
```
Comment (Aerdayne) [inline README.md:44]:
```suggestion
However, if you need a simple HTTP client for some ad-hoc one-off task, you can configure one using `.make`, without necessity to define a separate client class. Please note that this method should not be used for any clients that are actively being utilised in the business logic.
```

And swap this with the paragraph and code snippet above.

**Comment by Aerdayne** [comment]
I'd swap the 2 paragraphs, so that the first one is for inheritance-based method and the following is about the inline one. Feel free to disregard.

File: README.md (hunk 1 above)
Comment (shalvah-gs) [reply README.md]: Done. Swapped the examples to favour the class one.

Comment (Aerdayne) [reply README.md]: Thank you!

**Review by a-berg-gs — APPROVED** [review]

**Pull request merged**